raster-foundry/granary
# Assimilate model dashboard

## Overview

This PR:

- adopts all the code from https://github.com/jisantuc/granary-model-dashboard
- serves the Elm app from the api server
- adds building the Elm app and putting it in the right magic place to CI
- updates the README to include instructions for frontend development

### Checklist

- ~Any frontend changes are hidden behind a feature flag~ lol

## Demo

![image](https://user-images.githubusercontent.com/5702984/89694684-b2a14f00-d8ce-11ea-9328-bc8eb92d51e7.png)

## Testing Instructions

- Follow the Quick Setup instructions:

```bash
$ nvm use
$ ./scripts/setup
$ ./scripts/update
```

- start the database: `docker-compose up -d database`
- fire up your server: `./scripts/server`
- load the frontend app at http://localhost:8080/index.html
- grab the token from the database:
  - `./scripts/dbshell`
  - `select id from tokens;`
- enter the token and confirm that the app "works", such as it is right now
- read the README and think about what else you'd like to know

Closes #366
